zproject/backends.py

```diff
@@ -351,10 +352,15 @@ def authenticate(self, request: HttpRequest, *,
         user_profile = common_get_active_user(username, realm, return_data=return_data)
         if user_profile is None:
             return None
-        if user_profile.check_password(password):
-            if hasattr(request, 'session'):
-                request.session['needs_to_change_password'] = not check_password_strength(password)
-                return user_profile
+        try:
```

Comment:
this can be made nicer by starting with `is_password_correct = user_profile.check_password(password)` - then you only need to have that in the

Reply:
That looks great. Changed it. Thanks!